tests/src/Simple/HelloWasm/Program.cs

```diff
@@ -134,6 +134,14 @@ private static unsafe void Main(string[] args)
         Action virtualDelegate = tempObj.VirtualDelegateTarget;
         virtualDelegate();
 
+        var arrayTest = new BoxStubTest[] { new BoxStubTest { Value = "Hello" }, new BoxStubTest { Value = "Array" }, new BoxStubTest { Value = "Test" } };
```

Comment:
Can you please also add tests for types that are larger and smaller than pointers?

Reply:
I'm not sure whats going on, having a single array of a larger or smaller type isn't a problem. But once I add multiple arrays to the test all of the baseSizes in the array EETypes turns totally bogus in unmanaged code.

Reply:
I haven't tried debugging this yet, but it occurred to me that other parts of the framework surely have arrays with different element sizes that didn't break this test, so it seems more likely that something goes wrong in code generation for this function when you have multiple arrays with different sizes. You could confirm that by adding separate functions with separate array sizes.

Reply:
I'm not going to claim to understand whats going wrong in RhNewArray but I've made the call directly into RhpNewArray + a few tweaks and everything runs nicely now. Unless there is a problem in EETypePtr.ToPointer I don't see anywhere else it could have been getting messed up. The problem with that theory is that RhNewObject has no such problems and makes more extensive use of the EEType*.

Reply:
That actually sounds a lot like some of the bugs I fixed in #5298, but I agree there's no need to dig further into it. If RhpNewArray is working, let's go with it 😄